src/Go.Windows/GoBuildCommand.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using System.ComponentModel.Design;
+using System.Diagnostics;
+using System.Globalization;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Task = System.Threading.Tasks.Task;
+using Microsoft.VisualStudio.Text;
+using EnvDTE;
+
+namespace Go.Windows
+{
+    /// <summary>
+    /// Command handler
+    /// </summary>
+    internal sealed class GoBuildCommand
+    {
+        /// <summary>
+        /// Command ID.
+        /// </summary>
+        public const int CommandId = 0x0100;
+
+        /// <summary>
+        /// Command menu group (command set GUID).
+        /// </summary>
+        public static readonly Guid CommandSet = new Guid("13629496-9f13-45fd-96d9-b5ba4ac9c086");
+
+        /// <summary>
+        /// VS Package that provides this command, not null.
+        /// </summary>
+        private readonly AsyncPackage package;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="GoBuildCommand"/> class.
+        /// Adds our command handlers for menu (commands must exist in the command table file)
+        /// </summary>
+        /// <param name="package">Owner package, not null.</param>
+        /// <param name="commandService">Command service to add command to, not null.</param>
+        private GoBuildCommand(AsyncPackage package, OleMenuCommandService commandService)
+        {
+            this.package = package ?? throw new ArgumentNullException(nameof(package));
+            commandService = commandService ?? throw new ArgumentNullException(nameof(commandService));
+
+            var menuCommandID = new CommandID(CommandSet, CommandId);
+            MenuCommand menuItem = new MenuCommand(this.BuildGoFile, menuCommandID);
+            commandService.AddCommand(menuItem);
+        }
+
+        private void BuildGoFile(object sender, EventArgs e)
+        {
+            // Show a message box to prove we were here
+            VsShellUtilities.ShowMessageBox(
+                this.package,
+                "Run go build",
+                "Go Build",
+                OLEMSGICON.OLEMSGICON_INFO,
+                OLEMSGBUTTON.OLEMSGBUTTON_OK,
+                OLEMSGDEFBUTTON.OLEMSGDEFBUTTON_FIRST);
+
+            // Try to get document path
+            var dte = (ServiceProvider.GetServiceAsync(typeof(DTE))).Result as DTE;
```

Comment:
DTE is more or less deprecated, better way to get services is via [GlobalServiceProvider](https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.shell.serviceprovider.globalprovider?view=visualstudiosdk-2017) or by MEF importing SVsServiceProvider.

Reply:
Removed